src/main/java/teammates/common/datatransfer/attributes/EntityAttributes.java

```diff
@@ -57,4 +58,18 @@ public boolean isValid() {
      * e.g. trim strings
      */
     public abstract void sanitizeForSaving();
+
+    /**
+     * Check the list for empty nodes ("") and remove them if found.
+     * Return the list without empty nodes or empty list if all nodes are empty
+     */
+    public List<String> removeEmptyElements(List<String> errors) {
```

Comment:
removeEmptyElements -> getNonEmptyErrors? Probably not the best name, please use a better one if you think of one. The idea is that removeEmptyElements is at a lower level of abstraction, since its talking about what the underlying algorithm is doing. removeEmptyElements also implies that the function is able to handle any generic element types, but it only works for Strings.